src/common/meta/src/key/table_name.rs

```diff
@@ -147,62 +150,94 @@ impl TableNameManager {
-            .context(InvalidCatalogValueSnafu)
+                .transpose()? else {
+                return UnexpectedSnafu {
+                    err_msg: format!("compare_and_put expect None but failed with current value None, key: {key}, value: {value:?}"),
+                }.fail()
+            };
+            ensure!(
+                curr.table_id == table_id,
+                TableAlreadyExistedSnafu {
+                    table_id: curr.table_id
+                }
+            );
+        }
+        Ok(())
     }
 
-    // TODO(LFC): Remove this method when table metadata refactor is done.
-    pub async fn tables_old(&self, catalog: &str, schema: &str) -> Result<Vec<String>> {
-        let key = build_table_global_prefix(catalog, schema);
-        let req = RangeRequest::new().with_prefix(key.as_bytes());
-
-        let resp = self.kv_backend.range(req).await?;
+    /// Rename a TableNameKey to a new table name. Will check whether the TableNameValue matches the
+    /// `expected_table_id` first. Can be executed again if the first invocation is successful.
+    pub async fn rename(
+        &self,
+        key: TableNameKey<'_>,
+        expected_table_id: TableId,
+        new_table_name: &str,
+    ) -> Result<()> {
+        let new_key = TableNameKey::new(key.catalog, key.schema, new_table_name);
+
+        if let Some(value) = self.get(key).await? {
+            ensure!(
+                value.table_id == expected_table_id,
+                RenameTableSnafu {
+                    reason: format!(
+                        "the input table name '{}' and id '{expected_table_id}' not match",
+                        Into::<TableName>::into(key)
+                    ),
+                }
+            );
 
-        let mut table_names = Vec::with_capacity(resp.kvs.len());
-        for kv in resp.kvs {
-            let key = TableGlobalKey::parse(String::from_utf8_lossy(kv.key()))
-                .context(InvalidCatalogValueSnafu)?;
-            table_names.push(key.table_name);
+            let txn = Txn::new()
+                .when(vec![
+                    Compare::with_value(
+                        key.as_raw_key(),
+                        CompareOp::Equal,
+                        value.try_as_raw_value()?,
+                    ),
+                    Compare::with_not_exist_value(new_key.as_raw_key(), CompareOp::Equal),
+                ])
+                .and_then(vec![
+                    TxnOp::Delete(key.as_raw_key()),
+                    TxnOp::Put(new_key.as_raw_key(), value.try_as_raw_value()?),
+                ]);
+
+            let resp = self.kv_backend.txn(txn).await?;
+            ensure!(
+                resp.succeeded,
+                RenameTableSnafu {
+                    reason: format!("txn failed with response: {:?}", resp.responses)
+                }
+            );
+        } else {
+            let Some(value) = self.get(new_key).await? else {
```

Comment:
Better to add some comments to explain why the absence of a new key means that this table does not exist.